processing/src/main/java/org/apache/carbondata/processing/sort/sortdata/SortDataRows.java

```diff
@@ -270,7 +270,7 @@ private void writeDataToFile(Object[][] recordHolderList, int entryCountLocal, F
             } else if (dataType == DataTypes.SHORT) {
               stream.writeShort((Short) value);
             } else if (dataType == DataTypes.INT) {
-              stream.writeInt((Integer) value);
+              stream.writeInt(Integer.parseInt(value.toString()));
```

Comment:
Can you post the error message and describe what is the root cause of this bug?

Reply:
https://issues.apache.org/jira/browse/CARBONDATA-2016

Reply:
here is exception trace 18/01/10 16:17:12 ERROR CompactionResultSortProcessor: [Executor task launch worker-36][partitionID:customer1;queryID:15798380253871] Compaction failed: java.lang.Long cannot be cast to java.lang.Integer
java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
at org.apache.carbondata.processing.sort.sortdata.SortDataRows.writeDataToFile(SortDataRows.java:273)
at org.apache.carbondata.processing.sort.sortdata.SortDataRows.startSorting(SortDataRows.java:214)
at org.apache.carbondata.processing.merger.CompactionResultSortProcessor.processResult(CompactionResultSortProcessor.java:226)
at org.apache.carbondata.processing.merger.CompactionResultSortProcessor.execute(CompactionResultSortProcessor.java:159)
at org.apache.carbondata.spark.rdd.CarbonMergerRDD$$anon$1.<init>(CarbonMergerRDD.scala:234)
at org.apache.carbondata.spark.rdd.CarbonMergerRDD.internalCompute(CarbonMergerRDD.scala:81)

Reply:
But what is the root cause of this bug? I think your change (convert to string and then int) is like hiding this bug.

Reply:
@jackylyk ok i get your point you mean we should handle it at the base root level ok I am looking into it